test/SILGen/vtable_thunks.swift

```diff
@@ -304,6 +314,10 @@ class Noot : Aap {
 // CHECK:         #B.i3!1: {{.*}} : @$s13vtable_thunks1F{{[A-Z0-9a-z_]*}}FTV
 // CHECK:         #B.i4!1: {{.*}} : @$s13vtable_thunks1F{{[A-Z0-9a-z_]*}}F
 
+// CHECK-LABEL: sil_vtable H {
+// CHECK:  #G.foo!1: <T where T : Proto> (G) -> (T) -> () : @$s13vtable_thunks1H{{[A-Z0-9a-z_]*}}tlF [override]
```

Comment:
It looks like this is not a thunk as expected (it would have the FTV mangling if it were). You need to dig into some of the code in emitVTableThunk() to see why it's not deciding to emit a thunk here.

Reply:
Okay let me check. From a quick look, it seems like it's returning inside

```c++
if (!baseLessVisibleThanDerived &&
      M.Types.checkFunctionForABIDifferences(derivedInfo.SILFnType,
                                             overrideInfo.SILFnType)
      == TypeConverter::ABIDifference::Trivial)
    return SILVTable::Entry(base, implFn, implKind);
```

Reply:
@slavapestov So, in `emitVTableMethod`, we have a check:

```
if (!baseLessVisibleThanDerived &&
    M.Types.checkFunctionForABIDifferences(derivedInfo.SILFnType, 
                                           overrideInfo.SILFnType) 
    == TypeConverter::ABIDifference::Trivial) { ... }
```

We're returning a SILVTable::Entry here, which is why there's no thunk.

Reply:
I think we might need to do the same check here as the one in `requiresNewVTableEntry` and fall through to emit the thunk instead of returning.

Reply:
Yeah, that seems to work:

>#G.foo!1: <T where T : Proto> (G) -> (T) -> () : @$s13vtable_thunks1HC3foo3argyx_tlFAA1GCAdEyx_tAA5ProtoRzlFTV [override]	// vtable thunk for G.foo<A>(arg:) dispatching to H.foo<A>(arg:)